components/ConfirmPopup.vue

```diff
@@ -0,0 +1,152 @@
+<template>
+  <Popup :show-popup="display">
+    <div class="popup-delete">
+      <span class="title">{{ title }}</span>
+      <span class="description">
+        {{ description }}
+      </span>
+      <label v-if="hasToType" for="confirmation" class="confirmation-label">
+        <span>
+          To confirm your action, please type
+          <span class="confirmation-text">{{ confirmationText }}</span>
+          to continue
+        </span>
+      </label>
+      <input
+        v-if="hasToType"
+        id="confirmation"
+        v-model="confirmation_typed"
+        type="text"
+        placeholder="Type your username to continue"
```

Comment:
Confirmation popup placeholder shouldn't be bound to username?

Reply:
Let me fix that so something more generic

Reply:
Done